kernel/data-pipeline/dialect/postgresql/src/main/java/org/apache/shardingsphere/data/pipeline/postgresql/ingest/PostgreSQLWALDumper.java

```diff
@@ -86,13 +98,43 @@ protected void runBlocking() {
                     continue;
                 }
                 AbstractWALEvent event = decodingPlugin.decode(message, new PostgreSQLLogSequenceNumber(stream.getLastReceiveLSN()));
-                channel.pushRecord(walEventConverter.convert(event));
+                if (decodeWithTX) {
+                    processEventWithTX(event);
+                } else {
+                    processEventIgnoreTX(event);
+                }
             }
         } catch (final SQLException ex) {
             throw new IngestException(ex);
         }
     }
     
+    private void processEventWithTX(final AbstractWALEvent event) {
+        if (event instanceof BeginTXEvent) {
+            walEvents.clear();
+            return;
+        }
+        if (event instanceof AbstractRowEvent) {
+            walEvents.add(event);
+            return;
+        }
+        if (event instanceof CommitTXEvent) {
+            walEvents.add(event);
+            List<Record> records = new LinkedList<>();
+            for (AbstractWALEvent each : walEvents) {
+                records.add(walEventConverter.convert(each));
+            }
+            channel.pushRecords(records);
+        }
```

Comment:
Why `event` is put to `channel` in `OpenGaussWALDumper` but not here

Reply:
improved, keep the method same